pkg/gardenlet/controller/seed/seed.go

```diff
@@ -187,7 +187,14 @@ func (c *Controller) Run(ctx context.Context, workers int) {
 			seed.Labels = utils.MergeStringMaps(map[string]string{
 				v1beta1constants.GardenRole: v1beta1constants.GardenRoleSeed,
 			}, c.config.SeedConfig.Labels)
-			seed.Spec = c.config.SeedConfig.Seed.Spec
+
+			// Convert gardenlet config to an external version
+			cfg, err := confighelper.ConvertGardenletConfigurationExternal(c.config)
```

Comment:
Nit: Can you move this up, so it's outside of `CreateOrUpdate`?
It doesn't depend on the object fetched from the API server.

Reply:
Done